Signal/src/view controllers/CallViewController.swift

```diff
@@ -541,27 +630,87 @@ class CallViewController: UIViewController, CallObserver {
     // MARK: - CallObserver
 
     internal func stateDidChange(call: SignalCall, state: CallState) {
-        DispatchQueue.main.async {
-            Logger.info("\(self.TAG) new call status: \(state)")
-            self.updateCallUI(callState: state)
-        }
+        AssertIsOnMainThread()
+        Logger.info("\(self.TAG) new call status: \(state)")
+        self.updateCallUI(callState: state)
     }
 
     internal func hasVideoDidChange(call: SignalCall, hasVideo: Bool) {
-        DispatchQueue.main.async {
-            self.updateCallUI(callState: call.state)
-        }
+        AssertIsOnMainThread()
+        self.updateCallUI(callState: call.state)
     }
 
     internal func muteDidChange(call: SignalCall, isMuted: Bool) {
-        DispatchQueue.main.async {
-            self.updateCallUI(callState: call.state)
-        }
+        AssertIsOnMainThread()
+        self.updateCallUI(callState: call.state)
     }
 
     internal func speakerphoneDidChange(call: SignalCall, isEnabled: Bool) {
-        DispatchQueue.main.async {
-            self.updateCallUI(callState: call.state)
+        AssertIsOnMainThread()
+        self.updateCallUI(callState: call.state)
+    }
+
+    // MARK: - Video
+
+    internal func updateLocalVideoTrack(localVideoTrack: RTCVideoTrack?) {
+        AssertIsOnMainThread()
+        if self.localVideoTrack == localVideoTrack {
```

Comment:
Done. I've been wondering why `guard` exists.  It just formalizes conventional early-exit practice.